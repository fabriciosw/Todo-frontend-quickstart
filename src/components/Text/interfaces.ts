import React from 'react';

export interface IText {
  children: React.ReactNode;
  as:
    | 'p'
    | 'small'
    | 'b'
    | 'i'
    | 'em'
    | 'mark'
    | 'del'
    | 'ins'
    | 'sub'
    | 'sup'
    | 'span'
    | 'strong'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6';
  size?: string;
  weight?: 400 | 500 | 700;
  color?: string;
  className?: string;
}
