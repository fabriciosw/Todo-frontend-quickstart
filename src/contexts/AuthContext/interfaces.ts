import React from 'react';
import firebase from '../../config/firebase';

export interface IAuthContext {
  currentUser: firebase.User | null;
  signup: (displayName: string, email: string, password: string) => Promise<firebase.auth.UserCredential>;
  login: (email: string, password: string) => Promise<firebase.auth.UserCredential>;
  logout: () => Promise<void>;
  resetPassword: (email: string) => Promise<void>;
  updateEmail: (email: string) => Promise<void>;
  updatePassword: (email: string) => Promise<void>;
}

export interface IAuthProvider {
  children: React.ReactElement;
}
