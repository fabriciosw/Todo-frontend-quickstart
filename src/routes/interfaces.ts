import React from 'react';

export interface IRoute {
  path: string;
  component: React.FunctionComponent;
  public?: boolean;
  title: string;
}
