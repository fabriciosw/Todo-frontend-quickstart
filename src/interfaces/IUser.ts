export interface IUser {
  createdAt: Date;
  first_name: string;
  last_name: string;
  email: string;
  id: string;
}
