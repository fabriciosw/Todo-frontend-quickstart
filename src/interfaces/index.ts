export * from './IData';
export * from './IUser';
