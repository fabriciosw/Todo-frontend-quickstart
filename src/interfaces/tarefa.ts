export interface ITarefa {
  id: number;
  title: string;
  description: string;
  complete: boolean;
}
