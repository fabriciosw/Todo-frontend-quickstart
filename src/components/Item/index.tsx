import React, { useContext } from 'react';
import { ITarefa } from '../../interfaces/tarefa';
import Trash from '../../statics/images/Trash.png';
import finish from '../../statics/images/finish.jpg';
import { Tarefa } from '../../contexts/TaskContext/index';
import * as todoServices from '../../services/todoServices';

interface Props {
  tarefa: ITarefa;
}

export default function Item({ tarefa }: Props): JSX.Element {
  const { setTarefas } = useContext(Tarefa);

  return (
    <div className={`tarefa${tarefa.complete ? ' finalizado' : ''}`} key={tarefa.id}>
      <div className="dados">
        <p>{tarefa.title}</p>
        <span>{tarefa.description}</span>
      </div>
      <div className="opcoes">
        {!tarefa.complete && (
          <button type="button" onClick={() => todoServices.endTask(tarefa.id, setTarefas)} className="button_item">
            <img src={finish} alt="terminar" />
          </button>
        )}

        <button type="button" onClick={() => todoServices.deleteTask(tarefa.id, setTarefas)} className="button_item">
          <img src={Trash} alt="deletar" />
        </button>
      </div>
    </div>
  );
}
