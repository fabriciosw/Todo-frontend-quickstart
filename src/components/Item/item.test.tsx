import React from 'react';
import { render } from '@testing-library/react';
import Item from './index';
import '@testing-library/jest-dom';
import { ITarefa } from '../../interfaces/tarefa';

describe('Item component', () => {
  it('must match snapshot', () => {
    const task: ITarefa = {
      id: 1,
      title: 'testing task',
      description: 'testing description',
      complete: false,
    };
    const { container } = render(<Item tarefa={task} />);

    expect(container).toMatchSnapshot();
  });
});
