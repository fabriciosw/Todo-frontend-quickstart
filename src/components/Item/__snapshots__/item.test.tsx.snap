// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component must match snapshot 1`] = `
<div>
  <div
    class="tarefa"
  >
    <div
      class="dados"
    >
      <p>
        testing task
      </p>
      <span>
        testing description
      </span>
    </div>
    <div
      class="opcoes"
    >
      <img
        alt="terminar"
        src="finish.jpg"
      />
      <img
        alt="deletar"
        src="Trash.png"
      />
    </div>
  </div>
</div>
`;
